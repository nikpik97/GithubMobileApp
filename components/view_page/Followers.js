import React from 'react';
import { ScrollView, TouchableHighlight, Linking } from 'react-native';
import UserBlock from '../help/UserBlock';

const githubapi = require('../../api/github_api')

//Component to display Followers information

export default class Followers extends React.Component 
{
    render() {
        return (<h1>Followers Not Implemented</h1>);
    }
}