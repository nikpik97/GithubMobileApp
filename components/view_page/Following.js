import React from 'react';

const githubapi = require('../../api/github_api')


export default class Following extends React.Component 
{
    /**
     * This class is a compenent to handle the Following page.
     * It's currently not yet implemented.
     */
    render() {
        return (<h1>Following Not Implemented</h1>);
    }
}