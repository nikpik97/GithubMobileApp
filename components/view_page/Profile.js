import React from 'react' 
import { View, Image, ScrollView, TouchableHighlight, Dimensions } from 'react-native'
import ContentBlock from '../help/ContentBlock'

const githubapi = require('../../api/github_api')



export default class Profile extends React.Component {
    /**
     * This class is a component to display Profile information
     */
    constructor(props) {
        super(props);
        this.state = {
            'search':'nikpik97',
            'profileInfo': {
                'name': '',
                'username': '',
                'bio': '',
                'email': '',
                'website': '',
                'repositories': '',
                'followers': '',
                'following': '',
                'avatar': '',
                'profile_creation_date': ''
            }, 
            'screen': {
                'height': 10,
                'width': 0
            }
        };
        this.loadUserProfile();
    }

    loadUserProfile () {
        return githubapi.getProfileInfo(this.state.search)
                        .then(profileInfo => {
                            this.setState({'profileInfo': profileInfo});
        });
    }

    onPressRepositories() {
        this.props.navigation.navigate('Repositories', {search:JSON.stringify(this.state.search)});
    }

    onPressFollowers() {
        this.props.navigation.navigate('Followers', {search:JSON.stringify(this.state.search)});
    }
    
    onPressFollowing() {
        this.props.navigation.navigate('Following', {search:JSON.stringify(this.state.search)});
    }

    orientation = (orientation) => {
        this.setState({'screen': Dimensions.get('window')});
    }

    render() {
        return (
            <ScrollView onLayout={ this.orientation }>
                <View>
                    <Image
                        style={{width: Dimensions.get('window').width, height: Dimensions.get('window').width}}
                        source={ this.state.profileInfo.avatar }
                    />
                </View>
                <ContentBlock iconType='person' title='Name' content={this.state.profileInfo.name}/>
                <ContentBlock iconType='github' title='Username' content={this.state.profileInfo.username}/>
                <ContentBlock iconType='bio' title='Bio' content={this.state.profileInfo.bio}/>
                <ContentBlock iconType='email' title='Email' content={this.state.profileInfo.email}/>
                <ContentBlock iconType='website' title='Website' content={this.state.profileInfo.website}/>
                <TouchableHighlight onPress={() => this.onPressRepositories()} underlayColor='#FFAAEE'>
                    <ContentBlock iconType='repo' title='Repositories' content={this.state.profileInfo.repositories}/>
                </TouchableHighlight>
                <TouchableHighlight onPress={() => this.onPressFollowers()} underlayColor='#FFAAEE'>
                    <ContentBlock iconType='group' title='Followers' content={this.state.profileInfo.followers}/>
                </TouchableHighlight>
                <TouchableHighlight onPress={() => this.onPressFollowing()} underlayColor='#FFAAEE'>
                    <ContentBlock iconType='group' title='Following' content={this.state.profileInfo.following}/>
                </TouchableHighlight>
                <ContentBlock iconType='profile_creation_date' title='Profile creation date' content={this.state.profileInfo.profile_creation_date}/>
            </ScrollView>
        );
    }
}
