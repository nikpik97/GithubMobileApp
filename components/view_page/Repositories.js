import React from 'react';
import { Linking, ScrollView, TouchableHighlight } from 'react-native';
import ContentBlock from '../help/ContentBlock';

const githubapi = require('../../api/github_api')


export default class Repositories extends React.Component {
    /**
     * This class handles rendering of the Repositories page 
     */
    constructor(props) {
        super(props);
        this.state = {
            'repositories': []
        };
        this.loadUserProfile();
        if (this.props.navigation.state.params != null)
            alert(this.props.navigation.state.params.search)
    }

    loadUserProfile() {
        /**
         * This function calls graphql api code and populates itself with the resulting data
         */
        return githubapi.getRepoInfo('nikpik97')
                        .then(repoInfo => {
                            this.setState({'repositories': repoInfo});
                        });
    }

    onPressRepository(url) {
        /** 
         * This function handles the redirect the the repo's corresponding url.
         */
        Linking.canOpenURL(url)
               .then(supported => {
                    if (supported) {
                        Linking.openURL(url);
                    } 
                    else {
                        window.alert("Cannot open following URL: " + url);
                    }
        });
    }

    render() {
        /**
         * This function handles the rendering of the resulting html block.
         */
        return (
            <ScrollView>
                {this.state.repositories.map((repo, index) => (
                    <TouchableHighlight key={index} onPress={() => this.onPressRepository(repo.url)} underlayColor='#FFAAEE'>
                        <ContentBlock iconType='repo' title={repo.name} content={repo.owner + ' --- ' + repo.description}/>
                    </TouchableHighlight>
                ))}
            </ScrollView>
        );
    }
}
