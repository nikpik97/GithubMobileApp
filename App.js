import React from 'react';
import { createStackNavigator } from 'react-navigation-stack';
import { Feather } from '@expo/vector-icons';
import { createBottomTabNavigator } from 'react-navigation-tabs';
import { createAppContainer } from 'react-navigation';

import Profile from './components/view_page/Profile';
import Repositories from './components/view_page/Repositories';
import Following from './components/view_page/Following';
import Followers from './components/view_page/Followers';

const Tabs = createBottomTabNavigator(
  /**
   * This object represents the tab Navigator functionality
   */
  {
    Profile: {
      screen: Profile,
      navigationOptions: {
        tabBarLabel: 'Profile',
        tabBarIcon: ({ tintColor }) => (<Feather name="user" size={35} color={tintColor} />)
      }
    },
    Repositories: {
      screen: Repositories,
      navigationOptions: {
        tabBarLabel: 'Repositories',
        tabBarIcon: ({ tintColor }) => (<Feather name="user-check" size={35} color={tintColor} />)
      }
    },
    Following: {
      screen: Following,
      navigationOptions: {
        tabBarLabel: 'Following',
        tabBarIcon: ({ tintColor }) => (<Feather name="list" size={35} color={tintColor} />)
      }
    },
      Followers: {
        screen: Followers,
        navigationOptions: {
          tabBarLabel: 'Followers',
          tabBarIcon: ({ tintColor }) => (<Feather name="users" size={35} color={tintColor} />)
      }
    }
  }
);


const Navigation = createStackNavigator(
  /**
   * This object represents the stack Navigator functionality.
   * It basically shows the  correspoding page to the tab that was chosen
   * in the tab navigator.
   */
  {
    Github_App: { screen: Tabs },
  }, 
  {
    headerMode: 'screen'
  }
);

const AppNavigator = createAppContainer(Navigation); 

export default class App extends React.Component {
  /** 
   * This class handles the rendering of the main App 
   */
  constructor(props) {
      super(props)
      this.state = {};
  }
  render() {
    return (
      <AppNavigator />
    );
  }
}