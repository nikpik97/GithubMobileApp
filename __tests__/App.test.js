import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';
import Profile from '../components/view_page/Profile';
import Repositories from '../components/view_page/Repositories';
import Following from '../components/view_page/Following';
import Followers from '../components/view_page/Followers';

test('Profile renders correctly', () => {
  const tree = renderer.create(<Profile />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Repositories renders correctly', () => {
  const tree = renderer.create(<Repositories />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Following renders correctly', () => {
  const tree = renderer.create(<Following />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Followers renders correctly', () => {
  const tree = renderer.create(<Followers />).toJSON();
  expect(tree).toMatchSnapshot();
});