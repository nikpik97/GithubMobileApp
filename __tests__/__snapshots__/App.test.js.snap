// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Followers renders correctly 1`] = `
<h1>
  Followers Not Implemented
</h1>
`;

exports[`Following renders correctly 1`] = `
<h1>
  Following Not Implemented
</h1>
`;

exports[`Profile renders correctly 1`] = `
<RCTScrollView
  onLayout={[Function]}
>
  <View>
    <View>
      <Image
        source=""
        style={
          Object {
            "height": 750,
            "width": 750,
          }
        }
      />
    </View>
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 30,
              "padding": 10,
            }
          }
        >
          Name
        </Text>
      </View>
      <Text
        style={
          Object {
            "fontSize": 20,
            "padding": 10,
          }
        }
      >
        
      </Text>
    </View>
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 30,
              "padding": 10,
            }
          }
        >
          Username
        </Text>
      </View>
      <Text
        style={
          Object {
            "fontSize": 20,
            "padding": 10,
          }
        }
      >
        
      </Text>
    </View>
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 30,
              "padding": 10,
            }
          }
        >
          Bio
        </Text>
      </View>
      <Text
        style={
          Object {
            "fontSize": 20,
            "padding": 10,
          }
        }
      >
        
      </Text>
    </View>
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 30,
              "padding": 10,
            }
          }
        >
          Email
        </Text>
      </View>
      <Text
        style={
          Object {
            "fontSize": 20,
            "padding": 10,
          }
        }
      >
        
      </Text>
    </View>
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 30,
              "padding": 10,
            }
          }
        >
          Website
        </Text>
      </View>
      <Text
        style={
          Object {
            "fontSize": 20,
            "padding": 10,
          }
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 30,
                "padding": 10,
              }
            }
          >
            Repositories
          </Text>
        </View>
        <Text
          style={
            Object {
              "fontSize": 20,
              "padding": 10,
            }
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 30,
                "padding": 10,
              }
            }
          >
            Followers
          </Text>
        </View>
        <Text
          style={
            Object {
              "fontSize": 20,
              "padding": 10,
            }
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 30,
                "padding": 10,
              }
            }
          >
            Following
          </Text>
        </View>
        <Text
          style={
            Object {
              "fontSize": 20,
              "padding": 10,
            }
          }
        >
          
        </Text>
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 30,
              "padding": 10,
            }
          }
        >
          Profile creation date
        </Text>
      </View>
      <Text
        style={
          Object {
            "fontSize": 20,
            "padding": 10,
          }
        }
      >
        
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Repositories renders correctly 1`] = `
<RCTScrollView>
  <View />
</RCTScrollView>
`;
